src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/MapFriendlyLocationControllerTest.kt

```diff
@@ -82,5 +84,25 @@ internal class MapFriendlyLocationControllerTest(@Autowired private val wac: Web
             .andExpect { model { attributeHasFieldErrorCode("form", "locationName", "NotEmpty") } }
             .andExpect { view { name("/map-location") } }
             .andExpect { status { isOk } }
+
+    verify(service, never()).locationAlreadyExists(any(), any())
+    verify(service, never()).mapFriendlyLocation(any(), any(), any())
+  }
+
+  @Test
+  internal fun `map new location fails when duplicate location supplied supplied`() {
+    whenever(service.locationAlreadyExists(agencyId, "Duplicate location")).thenReturn(true)
+
+    mockMvc.post("/map-location") {
+      param("agencyId", agencyId)
+      param("locationName", "Duplicate location")
+      param("locationType", "CC")
+    }
+            .andExpect { model { attributeErrorCount("form", 1) } }
```

Comment:
Not ideal I would rather check the error message.  Could do with looking at a later date...